### PR TITLE
added initial drq_v2

### LINKED ISSUE #25: Implement DrQ-v2
@article{yarats2021mastering,
  title={Mastering visual continuous control: Improved data-augmented reinforcement learning},
  author={Yarats, Denis and Fergus, Rob and Lazaric, Alessandro and Pinto, Lerrel},
  journal={arXiv preprint arXiv:2107.09645},
  year={2021}
}

https://arxiv.org/pdf/2107.09645

### PR DESCRIPTION
**Pull request by TheGogy**
Added initial DrQ-v2 agent.

Closes #25

### CODE REVIEW

File: src/agents/common/shared_encoder.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+"""
+Shared encoder
```
Comment (Tom-Pecher) [inline src/agents/common/shared_encoder.py]: Should we specify as the shared encoder, if its in common isn't it automatically shared?

File: src/main.py [hunk 2]
```diff
@@ -41,6 +41,10 @@ def main():
             from agents.dqn.train import train_agent
             agent = train_agent(args.hardcore, args.render)
 
+        case 'drq_v2':
+            from agents.drq_v2.train import train_agent
+            agent = train_agent(args.hardcore, args.render)
+
         case 'ppo':
             from agents.ppo.train import train_agent
             agent = train_agent(args.hardcore, args.render)
```
Comment (Tom-Pecher) [inline src/main.py]: Line is repeated, move outside of match case?

**Review by Tom-Pecher — COMMENTED** [review]
Looks great, just a few questions.

File: src/agents/common/shared_encoder.py (hunk 1 above)
Comment (TheGogy) [reply src/agents/common/shared_encoder.py]: Made more sense to me at the time of implementation.
Though it is in the common folder so you're right

File: src/main.py (hunk 2 above)
Comment (TheGogy) [reply src/main.py]: Done

**Comment by TheGogy** [comment]
Alright, added everything, merging now

**Pull request merged**